src/App.tsx

```diff
@@ -1,24 +1,254 @@
+      })
+      .finally(() => {
+        setProcessingTodoIds(prevTodoIds => prevTodoIds
+          .filter(id => id !== todo.id));
+      });
+  };
+
+  const handleChangeStatusTodo = (todo: Todo) => {
+    setProcessingTodoIds(prevTodoids => [...prevTodoids, todo.id]);
+
+    todoService
+      .updateTodo({
+        id: todo.id,
+        title: todo.title,
+        userId: todo.userId,
+        completed: !todo.completed,
+      })
+      .then(updatedTodo => {
+        setTodos(prevState => prevState.map(currentTodo => {
+          return currentTodo.id !== updatedTodo.id ? currentTodo : updatedTodo;
+        }));
+      })
+      .catch(() => {
+        setErrorMessage('Unable to update a todo');
+      })
+      .finally(() => {
+        setProcessingTodoIds(prevTodoIds => prevTodoIds
+          .filter(id => id !== todo.id));
+      });
+  };
+
+  const handleClearCompletedTodos = () => {
+    todos
+      .filter(todo => todo.completed)
+      .forEach(todo => {
+        handleDeleteTodo(todo.id);
+      });
+  };
+
+  const visibleTodos = filterTodos(todos, status);
+  const handleFilterStatus = (todosFilter: Status) => (
+    setStatus(todosFilter));
+
+  const allTodosCount = todos.length;
+  const CompletedTodosCount = todos
+    .filter(todo => todo.completed === true).length;
+  const isAllTodosCompleted = allTodosCount === CompletedTodosCount;
+
+  const handleChangeStatusTodos = () => {
+    if (isAllTodosCompleted) {
+      todos
+        .forEach(todo => {
+          handleChangeStatusTodo(todo);
+        });
+    } else {
+      todos
+        .filter(todo => todo.completed === false)
+        .forEach(todo => {
+          handleChangeStatusTodo(todo);
+        });
+    }
+  };
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <TodoHeader
+          onTodoAdd={handleAddTodo}
+          onTodoAddError={setErrorMessage}
+          isAllTodosCompleted={isAllTodosCompleted}
+          onTodosChangeStatus={handleChangeStatusTodos}
+          todos={visibleTodos}
+          isTodosHere={Boolean(allTodosCount)}
+        />
+
+        <section className="todoapp__main" data-cy="TodoList">
+          {visibleTodos.map(todo => {
+            return (
+              <TodoRow
+                todo={todo}
+                key={todo.id}
+                onTodoDelete={() => handleDeleteTodo(todo.id)}
+                onTodoRename={(todoTitle) => handleRenameTodo(todo, todoTitle)}
+                onTodoChangeStatus={() => handleChangeStatusTodo(todo)}
```

Comment:
So do you really need all these functions in `App` component and then pass them through props
Maybe better to create them inside component where they are used